src/AdminConsole/Services/MagicLinks/MagicLinkBuilder.cs

```diff
@@ -9,14 +9,19 @@ namespace Passwordless.AdminConsole.Services.MagicLinks;
 
 public class MagicLinkBuilder(
     SignInManager<ConsoleAdmin> signInManager,
-    IPasswordlessClient passwordlessClient,
+    IServiceProvider services,
     IActionContextAccessor actionContextAccessor,
     IUrlHelperFactory urlHelperFactory,
     ILogger<MagicLinkBuilder> logger)
     : IMagicLinkBuilder
 {
     public async Task<string> GetLinkAsync(string email, string? returnUrl = null)
     {
+        // Passwordless client is fetched lazily because the MagicLinkBuilder may need to instantiate
+        // in a context where the API is not accessible, for example when the Admin Console has not
+        // been initialized yet. Doing this prevents it from crashing unless the client is actually used.
+        var passwordlessClient = services.GetRequiredService<IPasswordlessClient>();
```

Comment:
This is not clean but this was the easiest solution.

A better solution would be to provide some sort of `IPasswordlessClientProxy` that would implement the `IPasswordlessClient` contract, but would instead initialize the client lazily each time any method is called. I think if we do go this route, this proxy should be provided by the .NET SDK directly to avoid duplicating the client initialization logic, especially when it comes to Identity integration.

Let me know what you think.